fs-cache/src/cache.rs

```diff
@@ -0,0 +1,60 @@
+use crate::memory_limited_storage::MemoryLimitedStorage;
+use data_error::Result;
+use std::path::Path;
+
+pub struct Cache<K, V> {
+    storage: MemoryLimitedStorage<K, V>,
+}
+
+impl<K, V> Cache<K, V>
+where
+    K: Ord
+        + Clone
+        + serde::Serialize
+        + serde::de::DeserializeOwned
+        + std::fmt::Display
+        + std::hash::Hash
+        + std::str::FromStr,
+    V: Clone + serde::Serialize + serde::de::DeserializeOwned,
+{
+    pub fn new(
+        label: String,
+        path: &Path,
+        max_memory_bytes: usize,
+    ) -> Result<Self> {
+        log::debug!(
+            "{} cache initialized with {} bytes limit",
+            label,
+            max_memory_bytes
+        );
+        Ok(Self {
+            storage: MemoryLimitedStorage::new(label, path, max_memory_bytes)?,
+        })
+    }
+
+    pub fn get(&mut self, key: &K) -> Option<V> {
+        let result = self.storage.get(key);
+        log::debug!(
+            "{} cache: get key={} -> found={}",
+            self.storage.label(),
+            key,
+            result.is_some()
+        );
```

Comment:
`result.is_some()` returns true if the option is a `Some` value, so printing that isn’t very useful. We could just print `result` directly.

Reply:
Actually, values can be huge. Like 100KB image. We should not log them. We can add parameter `log_values` to the storage constructor though, so small values would be logged all the time (responsibility of app developer).

I would refactor logging here like this:
- `log::debug("cache/{}: retrieving value for key {}", ...)`
- `self.storage.get`
- `log::warn` if the result is `None`

Reply:
Added!